### PR TITLE
[RNMobile] Search Block: Basic UI

### PR DESCRIPTION
**Pull request by AmandaRiu**
## Description
This PR builds on two previous PRs https://github.com/WordPress/gutenberg/pull/28983 and https://github.com/WordPress/gutenberg/pull/28985 (~those two have to be merged before this one can be taken out of draft~ done!). This PR adds the block UI components and simple styling. Full styling won't be added until a later PR when designs have been finalized. Working features include:
- Displaying and customizing label text. 
- Displaying search input box and either the placeholder text or custom placeholder text.
- Displaying Search button.
- Search button text support and icon only modes now supported in the UI. 
- Hide search label now works

Not included:
- Button position changes do not yet work and will be coming in a future PR. 

https://user-images.githubusercontent.com/5810477/108113864-0b178d00-7066-11eb-9beb-2d51d69138ef.mov

### Gutenberg mobile PR
https://github.com/wordpress-mobile/gutenberg-mobile/pull/3186

#### Known issues
- Button text wrapping acts differently between iOS and Android. This will get addressed in a later PR. 
- Both the search label and search button text use `RichText` and supports formatting but for some odd reason color changes only seem to work in the WordPress app, and not in the demo. Example:

Demo | WordPress
-- | --
color-text-demo|color-text-wordpress

- Button placeholder text is partially hidden

Screen Shot 2021-02-16 at 2 52 55 PM

## How has this been tested?
Test the following scenarios and verify the component updates appropriately:
1. Hide and unhide search label by clicking on the option on the toolbar and verify.
2. Toggle icon button only and verify. 
3. Update search label text and verify.
4. update search button text and verify.
5. Update placeholder text and verify.
6. Remove search label text - placeholder should say "Add label..."
7. Remove button text - placeholder should say "Add button text..." (full text not displayed on iOS - known issue)
8. Remove search input placeholder, placeholder should say "Optional placeholder...". 
 

## Types of changes
Part of a new feature to port the Search block to mobile

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers.  (Will be handled in a different PR)
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/search/edit.native.js [hunk 1]
```diff
@@ -72,19 +113,81 @@ export default function SearchEdit( { attributes, setAttributes } ) {
...
 	return (
-		<View>
+		<View { ...blockProps } style={ styles.searchBlockContainer }>
 			{ controls }
 
 			{ showLabel && (
 				<RichText
+					className="wp-block-search__label"
+					style={ {
+						...styles.searchLabel,
```
Comment (cameronvoell) [inline packages/block-library/src/search/edit.native.js:170]: Nice use of [spread attributes](https://reactjs.org/docs/jsx-in-depth.html#spread-attributes) 👍

**Review by cameronvoell — APPROVED** [review]
Tested in WPAndroid, looks really good! I'll keep an eye out on the known issues and the button position changes in future PR. This looks and works great though 🎉

File: packages/block-library/src/search/edit.native.js [hunk 2]
```diff
@@ -1,30 +1,75 @@
...
 	// Temporary. Will be removed when styling is implemented
 	// in a future PR.
 	const alert = ( message ) => {
 		Alert.alert( '', message, [ { text: 'OK' } ], { cancelable: true } );
 	};
 
+	const getBlockClassNames = () => {
+		return classnames(
+			className,
+			'button-inside' === buttonPosition
```
Comment (enejb) [inline packages/block-library/src/search/edit.native.js:48]: I think some of this function could be simplified by creating a map 

Something like 
```
const buttonPostionToClassMap = {
 'button-inside' : 'wp-block-search__button-inside',
 'button-outside': 'wp-block-search__button-outside',
 'no-button': 'wp-block-search__no-button',
 'button-only': 'wp-block-search__button-only',
}
let noButtonClass = null;
if( 'no-button' !== buttonPosition ) {
 noButtonClass = buttonUseIcon ? 'wp-block-search__text-button' : 'wp-block-search__text-button';
}
return classnames(
	className,
	buttonPosition && buttonPostionToClassMap [ buttonPosition ],
	noButtonClass
);
```

Let me know what you think :)

Comment (AmandaRiu) [reply packages/block-library/src/search/edit.native.js:48]: Thanks for this @enejb ! This PR is already merged, but I have many more in the pipeline so I'll use this code in the future!

**Pull request merged**